### ISSUE TITLE
RELAX: Likelihood function evaluation encountered a NaN

### ISSUE THREAD

**Issue by gregoruar** [veg/hyphy#1010]
Hello, everyone! I installed HYPHY 2.5.0 and launched test with RELAX of the file with a codon alignment and nwk tree, all formated as in the tutorial.
A part of calculations was done right, but then process was aborted with the error. Can someone help me interpret the error and help going forward with the analysis?

```
>rates –> 3

>models –> All

### Obtaining branch lengths and nucleotide substitution biases under the nucleotide GTR model
* Log(L) = -541560.86, AIC-c = 1085039.78 (959 estimated parameters)

### Obtaining the global omega estimate based on relative GTR branch lengths and nucleotide substitution biases
* Log(L) = -537289.40, AIC-c = 1076512.94 (967 estimated parameters)
* non-synonymous/synonymous rate ratio for *Reference* =   0.6503
* non-synonymous/synonymous rate ratio for *Test* =   0.6537

### Improving branch lengths, nucleotide substitution biases, and global dN/dS ratios under a full codon model
* Log(L) = -537207.21, AIC-c = 1076348.57 (967 estimated parameters)
* non-synonymous/synonymous rate ratio for *Reference* =   0.6331
* non-synonymous/synonymous rate ratio for *Test* =   0.6401

### Fitting the general descriptive (separate k per branch) model

### * Log(L) = -529472.61, AIC-c = 1062787.78 (1921 estimated parameters)
* The following baseline rate distribution for branch-site combinations was inferred

|          Selection mode           |     dN/dS     |Proportion, %|               Notes               |
|-----------------------------------|---------------|-------------|-----------------------------------|
|        Negative selection         |     0.000     |    0.108    |                                   |
|        Negative selection         |     0.726     |   99.880    |                                   |
|      Diversifying selection       |   49994.202   |    0.012    |                                   |

* Branch-level relaxation or intensification parameter distribution has mean  2.23, median  1.32, and 95% of the weight in  0.39 -  6.32

### Fitting the alternative model to test K != 1
Error:

Master node received an error:Internal error 
Likelihood function evaluation encountered a NaN (probably due to a parameterization error or a bug).

Function call stack
1 :  [namespace = otTneAdM] Optimize(mles, likelihoodFunction);

-------
2 :  relax.alternative_model.fit=estimators.FitLF(relax.filter_names,relax.trees,{"0":relax.model_map},relax.general_descriptive.fit,relax.model_object_map,{terms.run_options.retain_lf_object:TRUE});
-------
3 :  relax.FitMainTestPair();
```
**EDIT** This error is reproducible: the file with the same codon alignment but other order of labeling in nwk tree, which was launched in parallel also terminated with the same error. Also, there was one difference between tutorial file and mine: I left length of the branches in the nwk tree. But it seems to me that it is not the reason behind this error.

**Comment by gregoruar** [comment]
Dear @spond,
Thank you for reaching out to me.
I have sent the file to you.

All the best,
Grigorii

**Comment by gregoruar** [comment]
Dear @spond,
Could you give an example of changed line to limit inferred values of ω?
I am lost a bit in the design of `BranchSiteREL.bf`.

All the best,
Grigorii

**Comment by spond** [comment]
Dear @gregoruar,

I was able to run your dataset through RELAX without issues using the current `dev` branch of Hyphy (which has a few changes relative to the 2.5.0 release). I recall making changes in the transition matrix computation routines to make them more robust to NaN and other cases, but it could also be an implementation issue. I am attaching the JSON file (as a `.gz`) with the results.  You can inspect it using [vision.hyphy.org/relax](http://
vision.hyphy.org/relax)

I'd be curious to see if the same `dev` branch works on your system.

Best,
Sergei

[codon_aln_and_labeled_tree_rif.fna.RELAX.json.gz](https://github.com/veg/hyphy/files/3560148/codon_aln_and_labeled_tree_rif.fna.RELAX.json.gz)

**Comment by gregoruar** [comment]
Dear @spond,

Thank you for your kind help! Interestingly enough, I decided to reinstall current `default` branch at first, and the analysis managed to cope up with high ω (last time it did not work on the same dataset). I suspect I previously installed older version, where the changes were not implemented.

**Comment by akaraw** [comment]
Hi @gregoruar ,

I am currently facing the same issue. Would kindly be able to let me know which version of hyphy that you managed to get to work? Is it available on bioconda?
I can not install dev version for some reason.

Cheers,
Anjana

**Comment by stevenweaver** [comment]
Dear @akaraw, 

The latest version of HyPhy (2.5.5) is on [bioconda](https://anaconda.org/bioconda/hyphy). What was in `dev` when this issue was opened was released in version `2.5.2`. 

As far as not being able to install from source, please make sure you're using a modern compiler and that the requirements specified [here](https://github.com/veg/hyphy#building-from-source) are met. 

Best,
Steven

**Comment by akaraw** [comment]
Hi @stevenweaver 

Thank you very much!
